crates/task-impls/src/consensus.rs

```diff
@@ -1274,12 +1332,29 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>, A: ConsensusApi<TYPES, I> +
                 error!("Failed to sign leaf.commit()!");
                 return false;
             };
+
+            let upgrade_cert = if self
+                .upgrade_cert
+                .as_ref()
+                .is_some_and(|cert| cert.view_number == view)
+            {
+                // If the cert view number matches, set upgrade_cert to self.upgrade_cert
+                // and set self.upgrade_cert to None.
+                //
+                // Note: the certificate is discarded, regardless of whether the vote on the proposal succeeds or not.
```

Comment:
I assumed this was the desired behavior, otherwise we can simply switch take() to clone()

Reply:
Yes, I think so! Just wanted to make it explicit.